carma-messenger-ui/website/scripts/carma.config.js

```diff
@@ -0,0 +1,61 @@
+/***
+* Source: https://www.kenneth-truyers.net/2013/04/27/javascript-namespaces-and-modules/
+
+***/
+
+'use strict';
+
+// Create the root namespace and making sure we're not overwriting it
+var CarmaJS = CarmaJS || {};
+
+// Create a general purpose namespace method
+// This will allow us to create namespace a bit easier
+CarmaJS.registerNamespace = function (namespace) {
+    var nsparts = namespace.split(".");
+    var parent = CarmaJS;
+
+    // we want to be able to include or exclude the root namespace
+    // So we strip it if it's in the namespace
+    if (nsparts[0] === "CarmaJS") {
+        nsparts = nsparts.slice(1);
+    }
+
+    // loop through the parts and create
+    // a nested namespace if necessary
+    for (var i = 0; i < nsparts.length; i++) {
+        var partname = nsparts[i];
+        // check if the current parent already has
+        // the namespace declared, if not create it
+        if (typeof parent[partname] === "undefined") {
+            parent[partname] = {};
+        }
+        // get a reference to the deepest element
+        // in the hierarchy so far
+        parent = parent[partname];
+    }
+    // the parent is now completely constructed
+    // with empty namespaces and can be used.
+    return parent;
+};
+
+
+CarmaJS.registerNamespace("CarmaJS.Config");
+
+CarmaJS.Config = (function () {
+        //Private variables
+        var ip = '127.0.0.1';//'192.168.88.10'; //'192.168.88.10'; 192.168.32.146;
```

Comment:
In order to keep this portable you need to use the carma.config.js file located in the respective configuration. 
That doesn't mean you have have this here, but we need to verify that the correct file is being loaded. 
Here is the config currently in development configuration
https://github.com/usdot-fhwa-stol/carma-messenger/blob/develop/carma-messenger-config/development/carma.config.js

Reply:
carma-messenger-ui/website/scripts/carma.config.js  file is loaded in my index.html file and is in use.
https://github.com/usdot-fhwa-stol/carma-messenger/blob/feature/ui_truck_inspection/carma-messenger-ui/website/index.html

I did a global search and find install.sh file contains carma.config.js, however it is commented out. I am not sure why  we need to carma.config.js in the development folder and leave it there. Please let me know if I can remove it.

Reply:
> For rosbridge.js, please add a space and a comment between line 70 and 71 (before "CarmaJS.WidgetFramework.closeWidgets();..). This just separates the logic a bit more to stand out.
> 
> ```
>         ...
>         document.getElementById('connected').style.display = 'inline';
>         
>         //After succesful connection to ROS, load the widget. 
>         CarmaJS.WidgetFramework.closeWidgets();
>         CarmaJS.WidgetFramework.loadWidgets();
>        ...
> ```

The changes are made and see commit ab3f58c